csrank/__init__.py

```diff
@@ -1,9 +1,16 @@
 __version__ = "1.1.0"
 
-from .choicefunction import *
-from .core import *
-from .dataset_reader import *
-from .discretechoice import *
-from .objectranking import *
+from .choicefunction import __all__ as choicefunction
+from .core import __all__ as core
+from .dataset_reader import __all__ as dataset_reader
+from .discretechoice import __all__ as discretechoice
+from .objectranking import __all__ as objectranking
 from .tunable import Tunable
 from .tuning import ParameterOptimizer
+
+__all__ = ["Tunable", "ParameterOptimizer"]
+__all__.extend(choicefunction)

```

Comment:
I'm unsure about this. We could also use `# noqa` to just skip the check here. Maybe we should reconsider whether re-exporting everything at the top level is even desirable.

Reply:
Also looks like the current solution is not working as expected:

```
ImportError while importing test module '/home/travis/build/kiudee/cs-ranking/csrank/tests/test_choice_functions.py'.

Hint: make sure your test modules/packages have valid Python names.

Traceback:

csrank/tests/test_choice_functions.py:27: in <module>

    from csrank.tests.test_ranking import check_params_tunable

csrank/tests/test_ranking.py:8: in <module>

    from csrank import PairedCombinatorialLogit

E   ImportError: cannot import name 'PairedCombinatorialLogit' from 'csrank' (/home/travis/build/kiudee/cs-ranking/csrank/__init__.py)
```

Reply:
Regarding the second comment: I think in tests it is better to import using the full path anyway.

Reply:
As discussed on slack, we might want to reduce the re-exports in the future but this is not the place to do it. So I'm going with `# noqa` comments for now.

Reply:
Also see https://github.com/kiudee/cs-ranking/issues/105.